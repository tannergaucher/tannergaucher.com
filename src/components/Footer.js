import React from 'react'
import styled from 'styled-components'
import { StaticQuery, graphql } from 'gatsby'
import PropTypes from 'prop-types'
import Link from '../components/styles/Link'

const StyledFooter = styled.div`
  height: 10px;
  background: grey;
`

const Footer = () => <StyledFooter />

export default Footer
