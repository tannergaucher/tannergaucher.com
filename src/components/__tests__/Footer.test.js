import React from 'react'
import renderer from 'react-test-renderer'
import { PureFooter as Footer } from '../Footer'

/* 
Problem: How to test components that use gatsby <Link /> ?

Solution: 

*/

describe(`<Footer/>`, () => {
  it(`renders correctly and matches the snapshot`, () => {
    const data = {
      site: {
        siteMetadata: {
          title: 'Tanner Gaucher',
        },
      },
    }

    const tree = renderer.create(<Footer data={data} />).toJSON()
    console.debug(<Footer />)
    expect(tree).toMatchSnapshot()
  })
})
