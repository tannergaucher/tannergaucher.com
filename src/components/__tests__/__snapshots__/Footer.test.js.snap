// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer/> renders correctly and matches the snapshot 1`] = `
<MockedLink
  className="sc-bdVaJa fCGAPp"
  fontSize={
    Array [
      1,
    ]
  }
  mb={2}
  none="true"
  style={
    Object {
      "alignSelf": "center",
    }
  }
  to="/"
>
  <div
    className="sc-uJMKN eIQQgj sc-VigVT jCvPJb"
  >
    <span
      aria-label="technologist emoji"
      role="img"
    >
      👨‍💻
    </span>
    Tanner Gaucher
  </div>
</MockedLink>
`;
